alphadia/workflow/peptidecentric/extraction_handler.py

```diff
@@ -333,75 +462,226 @@ def _score_candidates(
                 # 'kernel_size': 20,  # 15?
+                "top_k_fragments": self._config["search"]["top_k_fragments_selection"],
                 "peak_length": self._config["search"]["quant_window"],
                 "mass_tolerance": self._optimization_manager.ms2_error,
                 "rt_tolerance": self._optimization_manager.rt_error,
                 "candidate_count": self._optimization_manager.num_candidates,
             }
         )
 
-        self._reporter.log_string(
-            f"Using parameters: fwhm_rt={scoring_params.fwhm_rt}, "
-            f"kernel_size={scoring_params.kernel_size}, "
-            f"peak_length={scoring_params.peak_length}, "
-            f"mass_tolerance={scoring_params.mass_tolerance}, "
-            f"rt_tolerance={scoring_params.rt_tolerance}"
+        candidates = PeakGroupSelection(selection_params).search(
+            dia_data, self._speclib_ng
+        )
+
+        cands = parse_candidates(candidates, spectral_library, dia_data)
+
+        return cands
+
+    def score_candidates(
+        self,
+        candidates_df: pd.DataFrame,
+        dia_data: "DiaDataNG",  # noqa: F821
+        spectral_library: SpecLibFlat,
+    ) -> pd.DataFrame:
+        """Score candidates using NG backend.
+
+        See superclass documentation for interface details.
+        """
+        self._lazy_init_speclib_ng(spectral_library)
+
+        candidates = candidates_to_ng(candidates_df, dia_data)
+
+        scoring_params = ScoringParameters()
+        scoring_params.update(
+            {
+                "top_k_fragments": self._config["search"]["top_k_fragments_scoring"],
+                "mass_tolerance": self._optimization_manager.ms2_error,
+            }
         )
 
-        peak_group_scoring = PeakGroupSelection(scoring_params)
+        candidate_features = PeakGroupScoring(scoring_params).score(
+            dia_data, self._speclib_ng, candidates
+        )
 
-        candidates = peak_group_scoring.search(dia_data_ng, speclib_ng)
+        features_df = to_features_df(candidate_features, spectral_library)
 
-        return parse_candidates(dia_data_, candidates, spectral_library.precursor_df)
+        return features_df
 
-    def _score_candidates(
+    def quantify_candidates(
         self,
         candidates_df: pd.DataFrame,
-        dia_data: tuple[DiaData, "DiaDataNG"],  # noqa: F821
+        precursor_fdr_df: pd.DataFrame | None,
+        dia_data: "DiaDataNG",  # noqa: F821
         spectral_library: SpecLibFlat,
+        top_k_fragments: int | None = None,
+    ) -> tuple[pd.DataFrame, pd.DataFrame]:
+        """Quantify candidates using NG backend.
+
+        See superclass documentation for interface details.
+        """
+        del top_k_fragments  # unused
+
+        self._lazy_init_speclib_ng(spectral_library)
+
+        candidates_collection = candidates_to_ng(candidates_df, dia_data)
+
+        # run quantification
+        quant_params = QuantificationParameters()
```

Comment:
don't we want to set the `top_k_fragments` override here? 
If not, then I would suggest to remove it from the function interface.